src/Utility/Security.php

```diff
@@ -121,17 +121,35 @@ public static function randomBytes(int $length): string
      * Creates a secure random string.
      *
      * @param int $length String length. Default 64.
+     * @param bool $inclSpecialChars If include special characters. Default false.
      * @return string
      */
-    public static function randomString(int $length = 64): string
+    public static function randomString(int $length = 64, bool $inclSpecialChars = false): string
     {
-        return substr(
-            bin2hex(Security::randomBytes((int)ceil($length / 2))),
+        $string = '';
+
+        if ($length === 0) {
+            return $string;
+        }
+
+        if ($inclSpecialChars) {
+            $special_charset = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~';
+
+            for ($i = 0; $i < rand(1, $length); $i++) {
+                $string .= $special_charset[rand(0, strlen($special_charset))];
```

Comment:
We can't use rand() in this context as it isn't a cryptographic secure source of entropy. The intent behind `randomString` is that is safe to use in a security minded context.

Reply:
Replaced rand() by cryptographically secure random_int()